enterprise/models.py

```diff
@@ -2,42 +2,238 @@
+    # for related field access. Although False is default value, it still makes sense to set it explicitly
+    # https://docs.djangoproject.com/en/1.10/topics/db/managers/#base-managers
+    use_for_related_fields = False
+
+    def get_queryset(self):
+        """
+        Return a new QuerySet object. Filters out inactive Enterprise Customers.
+        """
+        return super(EnterpriseCustomerManager, self).get_queryset().filter(active=True)
+
+
 @python_2_unicode_compatible
 class EnterpriseCustomer(TimeStampedModel):
     """
-    TODO: replace with a brief description of the model.
+    Enterprise Customer is an organization or a group of people that "consumes" courses.
+
+    Members of Enterprise Customer take courses on the edX platform.
+
+    Enterprise Customer might be providing certain benefits to their members, like discounts to paid course
+    enrollments, and also might request (or require) sharing learner results with them.
+
+    Fields:
+        [implicit] id (IntegerField): EnterpriseCustomer identifier, PRIMARY KEY
+        name (CharField): Enterprise Customer name
+        short_code (AutoSlugField): Enterprise Customer code - used to reference this Enterprise Customer in other
+                                    parts of the system (SSO, ecommerce, analytics etc.)
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        sso_providers (CharField): list of comma-separated SSO provider slugs. Providers in this list should link
+                                    users registered with that provider with Enterprise Customer
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
+        members (ManyToMany[User]): list of Enterprise Customer members.
     """
 
-    # TODO: add field definitions
+    NO_DATA_SHARING = "no_sharing"
+    REQUEST_DATA_SHARING = "request"
+    REQUIRE_DATA_SHARING = "require"
+
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomer model.
+        """
+
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer"
+        verbose_name_plural = "Enterprise Customers"
+
+    objects = EnterpriseCustomerManager()
+    all_objects = models.Manager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_("Enterprise Customer name."))
+    short_code = AutoSlugField(
+        max_length=120, blank=False, null=False,
+        populate_from="name", unique=True, editable=False,
+        help_text=_("Short code will be used to reference this Enterprise Customer in other parts of the system,"
+                    "i.e. SSO, ecommerce app, analytics etc.")
+    )
```

Comment:
nit: When adding a new EnterpriseCustomer it currently isn't possible to override the default value for `short_code`, it always generates it from the name. It would be nice if it was possible to provide a custom value for `short_code`.